internal/encryption/keyprovider.go

```diff
@@ -120,79 +188,32 @@ func (e *targetBuilder) setupKeyProvider(cfg config.KeyProviderConfig, stack []c
 
 	// Locate all the dependencies
 	deps, varDiags := gohcl.VariablesInBody(cfg.Body, keyProviderConfig)
-	diags = append(diags, varDiags...)
+	diags = diags.Extend(varDiags)
```

Comment:
FWIW this `Extend` method is exactly equivalent to the `append` you replaced. I don't have a strong opinion about which way to write it so this is just FYI, but other parts of the system do tend to just use `append` directly. :man_shrugging:

Reply:
I got a bit carried away, but in general I like the Extend / Append methods on hcl diags a bit more than the append syntax.